mlflow/tensorflow/__init__.py

```diff
@@ -887,14 +897,113 @@ def _patch_implementation(
                     # & tf.keras callbacks if necessary
                     callbacks = list(kwargs.get("callbacks") or [])
                     kwargs["callbacks"], self.log_dir = _setup_callbacks(
-                        callbacks, log_models, metrics_logger
+                        callbacks, False, metrics_logger
                     )
 
                 early_stop_callback = _get_early_stop_callback(callbacks)
                 _log_early_stop_callback_params(early_stop_callback)
 
                 history = original(inst, *args, **kwargs)
 
+                if log_models:
+                    import numpy as np
+
+                    def _is_x_type_supported_for_autologging(input_data):
+                        is_x_supported = False
+                        if isinstance(input_data, np.ndarray):
+                            is_x_supported = True
+                        elif isinstance(input_data, pandas.Series):
+                            is_x_supported = True
+                        elif isinstance(input_data, tensorflow.data.Dataset):
+                            is_x_supported = True
+                        elif isinstance(input_data, dict):
+                            is_x_supported = True
+                            for name in input_data.keys():
+                                data_type = input_data[name]
+                                if not isinstance(data_type, np.ndarray):
+                                    is_x_supported = False
+                        elif isinstance(input_data, tensorflow.keras.utils.Sequence):
+                            if isinstance(input_data[:][0], np.ndarray):
+                                is_x_supported = True
+
+                        if not is_x_supported:
+                            warnings.warn(
+                                "Tensorflow keras autologging only "
+                                "supports one of these types: numpy array, "
+                                "dictionary of (name -> numpy.ndarray), "
+                                "tf.data dataset or tf.keras.utils.Sequence"
+                            )
+                        return is_x_supported
+
+                    def _get_input_data_slice(input_example, history):
```

Comment:
does this shadow the higher level `history` variable? couldn't we just refer to it direclty with a history.params.get("steps", None) and then do a validate on the existence of that integer from within this private scoped function?

Reply:
good point, fixed. thanks.